adapters/between/between.go

```diff
@@ -3,169 +3,219 @@ package between
+}
 
-	if response.StatusCode == http.StatusBadRequest {
-		return nil, []error{&errortypes.BadInput{
-			Message: fmt.Sprintf("Unexpected status code: %d. Bad request to dsp", response.StatusCode),
-		}}
-	} else if response.StatusCode != http.StatusOK {
-		return nil, []error{&errortypes.BadServerResponse{
-			Message: fmt.Sprintf("ERR, response with status %d", response.StatusCode),
-		}}
+func buildImpBanner(imp *openrtb.Imp) error {
+	if imp.Banner == nil {
+		return &errortypes.BadInput{
+			Message: fmt.Sprintf("Request needs to include a Banner object"),
+		}
 	}
 
-	var bidResp openrtb.BidResponse
+	bannerCopy := *imp.Banner
+	banner := &bannerCopy
 
-	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
-		return nil, []error{err}
+	if banner.W == nil && banner.H == nil {
+		if len(banner.Format) == 0 {
+			return &errortypes.BadInput{
+				Message: fmt.Sprintf("Need at least one size to build request"),
+			}
+		}
+		format := banner.Format[0]
+		banner.Format = banner.Format[1:]
+		banner.W = &format.W
+		banner.H = &format.H
+		imp.Banner = banner
 	}
 
-	adapterResponse := adapters.NewBidderResponse()
-	adapterResponse.Currency = bidResp.Cur
+	return nil
+}
 
-	for _, seatBid := range bidResp.SeatBid {
-		for i := 0; i < len(seatBid.Bid); i++ {
-			bid := seatBid.Bid[i]
-			adapterResponse.Bids = append(adapterResponse.Bids, &adapters.TypedBid{
-				Bid:     &bid,
-				BidType: getMediaType(bid.ImpID, internalRequest.Imp),
-			})
-		}
+// Add Between required properties to Imp object
+func addImpProps(imp *openrtb.Imp, secure *int8, betweenExt *openrtb_ext.ExtImpBetween) {
+	imp.Secure = secure
+	if bidFloor, err := strconv.ParseFloat(betweenExt.BidFloor, 64); err != nil || bidFloor <= 0 {
+		imp.BidFloor = defaultBidfloor
+	} else {
+		imp.BidFloor = bidFloor
 	}
+}
 
-	return adapterResponse, nil
+// Adding header fields to request header
+func addHeaderIfNonEmpty(headers http.Header, headerName string, headerValue string) {
+	if len(headerValue) > 0 {
+		headers.Add(headerName, headerValue)
+	}
 }
 
-func splitImpressions(imps []openrtb.Imp) (map[openrtb_ext.ExtImpBetween][]openrtb.Imp, []error) {
-	var errors = make([]error, 0)
-	var m = make(map[openrtb_ext.ExtImpBetween][]openrtb.Imp)
+// Handle request errors and formatting to be sent to Between
+func preprocess(request *openrtb.BidRequest) (*openrtb_ext.ExtImpBetween, []error) {
+	errors := make([]error, 0, len(request.Imp))
+	resImps := make([]openrtb.Imp, 0, len(request.Imp))
+	secure := int8(0)
 
-	for _, imp := range imps {
-		bidderParams, err := getBidderParams(&imp)
+	if request.Site != nil && request.Site.Page != "" {
+		pageURL, err := url.Parse(request.Site.Page)
+		if err == nil && pageURL.Scheme == "https" {
+			secure = int8(1)
+		}
+	}
+
+	var betweenExt *openrtb_ext.ExtImpBetween
+	for _, imp := range request.Imp {
+		var err error
+		betweenExt, err = unpackImpExt(&imp)
```

Comment:
This would mean that you only consider the `ext` of the last imp in the request and use the values set in that to build the endpoint URL. So if the first imp has say an `ext` with `host` set as `127.0.0.1` and the next one has `host` set as `127.0.0.2` then you'd consider `127.0.0.2` as the host when building the URL. Are you sure you're okay with this behavior?